jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/JMSConnectionFactory.java

```diff
@@ -571,9 +571,9 @@ private Destination createDestination(Session session, String destinationName) t
              */
             try {
                 if (this.destinationType.equals(JMSConstants.JMSDestinationType.QUEUE)) {
```

Comment:
This may produce NPE. Please change this if condition other way (i.e SomeConstant equals actual value)

Reply:
Fixed with 77f8d26